lib/Helper.php

```diff
@@ -273,45 +273,54 @@ public static function warn( $message ) {
 	/**
 	 * Triggers a deprecation warning.
 	 *
+	 * If you want to catch errors like these in tests, then add the @expectedDeprecated tag to the
+	 * DocBlock. E.g.: "@expectedDeprecated {{ TimberImage() }}".
+	 *
 	 * @api
+	 *
 	 * @param string $function    The name of the deprecated function/method.
 	 * @param string $replacement Function to use instead.
 	 * @param string $version     When we deprecated this.
+	 *
 	 * @return void
 	 */
 	public static function deprecated( $function, $replacement, $version ) {
 		if ( ! WP_DEBUG ) {
 			return;
 		}
 
-		 do_action( 'deprecated_function_run', $function, $replacement, $version );
-
-	    /**
-	     * Filters whether to trigger an error for deprecated functions.
-	     *
-	     * @since 2.5.0
-	     *
-	     * @param bool $trigger Whether to trigger the error for deprecated functions. Default true.
-	     */
-	    if ( WP_DEBUG && apply_filters( 'deprecated_function_trigger_error', true ) ) {
-	        if ( function_exists( '__' ) ) {
-	            if ( ! is_null( $replacement ) ) {
-	                /* translators: 1: PHP function name, 2: version number, 3: alternative function name */
-	                trigger_error( sprintf( __('%1$s is <strong>deprecated</strong> since Timber version %2$s! Use %3$s instead.'), $function, $version, $replacement ) );
-	            } else {
-	                /* translators: 1: PHP function name, 2: version number */
-	                trigger_error( sprintf( __('%1$s is <strong>deprecated</strong> since Timber version %2$s with no alternative available.'), $function, $version ) );
-	            }
-	        } else {
-	            if ( ! is_null( $replacement ) ) {
-	                trigger_error( sprintf( '%1$s is <strong>deprecated</strong> since Timber version %2$s! Use %3$s instead.', $function, $version, $replacement ) );
-	            } else {
-	                trigger_error( sprintf( '%1$s is <strong>deprecated</strong> since Timber version %2$s with no alternative available.', $function, $version ) );
-	            }
-	        }
-	    }
-	}
+		do_action( 'deprecated_function_run', $function, $replacement, $version );
+
+		/**
+		 * Filters whether to trigger an error for deprecated functions.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param bool $trigger Whether to trigger the error for deprecated functions. Default true.
+		 */
+		if ( apply_filters( 'deprecated_function_trigger_error', true ) ) {
+			return;
+		}
+
+		if ( ! is_null( $replacement ) ) {
+			$error_message = sprintf(
+				'%1$s is <strong>deprecated</strong> since Timber version %2$s! Use %3$s instead.',
+				$function,
+				$version,
+				$replacement
+			);
+		} else {
+			$error_message = sprintf(
+				'%1$s is <strong>deprecated</strong> since Timber version %2$s with no alternative available.',
+				$function,
+				$version
+			);
+		}
 
+		// phpcs:disable WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+		// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
This all looks much cleaner and readable. My only Q is these two lines — are they necessary for something you're using with PHP_CodeSniffer? or just accidentally left in?

Reply:
Yes, I left them there deliberately. As described in https://github.com/timber/timber/blob/master/CONTRIBUTING.md#use-php_codesniffer-to-detect-coding-standard-violations, we are trying to stay as close possible to the standards. However, sometimes we conciously do something that wouldn’t be allowed under the standards by whitelisting certain CodeSniffer rules.

By adding these comments, CodeSniffer will not flag the issue *and* we have some kind of documentation for where we deviate from the standards.